homeassistant/components/alarmdecoder/binary_sensor.py

```diff
@@ -88,6 +88,10 @@ def __init__(
             CONF_ZONE_NUMBER: self._zone_number,
         }
 
+        # For RFID sensors we have a serial number which can be used as a unique id
+        if zone_rfid is not None:
+            self._attr_unique_id = "alarmdecoder_" + zone_rfid
```

Comment:
```suggestion
            self._attr_unique_id = zone_rfid
```

unique ids are per integration and per platform so no need to prefix it here.

Reply:
Is the rfid globally unique or is it possible to have two systems with the same rfid?

https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique&_highlight=id#example-acceptable-sources-for-a-unique-id

Reply:
It should meet those requirements, but I'll add a caveat as you made me learn of an extra feature in the integration I didn't know about where the unique ID should be zone_rfid + "_" + zone_loop because the way the integration works is I can define a different entity per loop within the sensor.

The `zone_rfid` value is a serial number printed on the device by Honeywell and is in the firmware of the individual sensor. When the sensor broadcasts it's current status it includes that serial number in the RFX message to say "I am device 1234567 and my status is open" so it is intended to be globally unique. I can't 100% say Honeywell never reused an ID obviously, but they're meant to be a device serial number that cannot be changed. That is why I'm limiting this change to just RF sensors. Hardwired sensors are only unique within your system, like it might just be "zone 3" because it's wired up to the contact for 3 on your PCB. RF sensors are different though because they have this serial number.

Second thing I'm curious of your input on. This integration is a little unique to me because most integrations add entities by like making an API call an getting the list of sensors. There is no such capability for Honeywell systems. So in the options flow you basically say "add a new sensor" and then type in the details. Since the zone_rfid+loop value is intended (and must be) unique, I would think I should also update the config_flow code to actually have validations to prevent the user from adding duplicates? Any precedent for something like this that you're aware of?

Reply:
I adjusted the unique id to guarantee uniqueness now but I'd love your thoughts on the config_flow side